### PR TITLE
Issue 42 - Grenade entities not destoryed

### PR DESCRIPTION
**Pull request by markus-wa**
See #42 

@micvbang please check if this helps.

I had to do some more complicated code than you suggested but this should catch everything, hopefully.

### CODE REVIEW

File: datatables.go [hunk 1]
```diff
@@ -285,11 +285,14 @@ func (p *Parser) bindGrenadeProjectiles(entity *st.Entity) {
 	})
 
 	entity.OnDestroy(func() {
-		p.eventDispatcher.Dispatch(events.NadeProjectileDestroyedEvent{
-			Projectile: proj,
-		})
+		// Issue #42
+		// Sometimes entities are destoyed even after round_officially_ended, sometimes not
+		// Due to that we need to check if we already deleted it, otherwise we get a nil pointer dereference
```
Comment (micvbang) [inline datatables.go]: Consider re-reading this comment :slightly_smiling_face:

File: datatables.go [hunk 2]
```diff
@@ -383,10 +395,14 @@ func (p *Parser) bindNewInferno(entity *st.Entity) {
 	})
 
 	entity.OnDestroy(func() {
-		p.eventDispatcher.Dispatch(events.InfernoExpiredEvent{
-			Inferno: inf,
-		})
-		delete(p.gameState.infernos, entityID)
+		// Issue #42
+		// Sometimes entities are destoyed even after round_officially_ended, sometimes not
+		// Due to that we need to check if we already deleted it, otherwise we get a nil pointer dereference
```
Comment (micvbang) [inline datatables.go]: Same as above comment

File: game_events.go [hunk 3]
```diff
@@ -79,7 +79,22 @@ func (p *Parser) handleGameEvent(ge *msg.CSVCMsg_GameEvent) {
...
 
-	case "round_officially_ended": // Round ended. . . probably the event where you get teleported to the spawn (=> You can still walk around between round_end and this?)
+	case "round_officially_ended": // The event after which you get teleported to the spawn (=> You can still walk around between round_end and this event)
+		// Issue #42
+		// Sometimes grenades & infernos aren't deleted / destroyed via entity-updates at the end of the round,
+		// so we need to do it here for those that weren't.
+		//
+		// We're not deleting them from entitites though as that's supposed to be as close to the actual demo data as possible.
+		// We're also not using Entity.Destroy() because it would - in some cases - be called twice on the same entity
+		// and it's supposed to be called when the demo actually says so (same case as with GameState.entities).
```
Comment (micvbang) [inline game_events.go]: I quite agree with your reasoning here.

File: datatables.go (hunk 1 above)
Comment (markus-wa) [reply datatables.go]: Sorry, I'm not quite following you here? Could you elaborate?

Comment (micvbang) [reply datatables.go]: Maybe it's just me, but I find that it's too general for what it actually covers.

Maybe something like the following?
> Sometimes grenade projectile entities are automatically destroyed after round_officially_ended, and sometimes they are not.
> The below check accommodates for the times they are not

Comment (markus-wa) [reply datatables.go]: > The below check accommodates for the times they are not.
Actually the opposite! :slightly_smiling_face: but I've changed it to hopefully be more clear.

**Pull request merged**